streams/lms_metadata.py

```diff
@@ -113,29 +119,26 @@ def connect(self):
             meta["track"] = song_data["title"]
             meta["image_url"] = song_data["artwork_url"]
         # File locking so to reduce errors, without locks here and on the read cycle you can sometimes read while writing, which will read an empty file and crash the stream
-        f = open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8')
-        try:
-          fcntl.flock(f, fcntl.LOCK_EX)
-          json.dump(meta, f, indent = 2)
-        finally:
-          fcntl.flock(f, fcntl.LOCK_UN)
-          f.close()
+        with open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8') as f:
```

Comment:
Using flock here will make the info process block and cause the server to have delays. To avoid this read/write problem it looks like the standard way is to write to a temporary file and then move the temporary file to the actual file name. When this happens the reading process continues reading the old file until it is closed.